src/components/Form/ToggleButton.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react';
+import { HtmlButton } from '../../reset';
+import { ToggleProps, baseClassName, ToggleState } from './Toggle';
+import { logger } from '../../utils/logger';
+
+export class ToggleButton extends Component<ToggleProps> {
+  state: ToggleState = {
+    toggleState: !!this.props.checked || !!this.props.defaultChecked || false,
+  };
+
+  handleClick = () => {
+    const { checked, onClick } = this.props;
+    const { toggleState } = this.state;
+    if (checked === undefined) {
+      this.setState({ toggleState: !toggleState });
+    }
+    if (!!onClick) {
+      onClick({ toggleState: !toggleState });
+    }
+  };
+
+  render() {
+    const {
+      disabled = false,
+      children,
+      checked: dissMissChecked,
+      defaultChecked: dissMissDefaultChecked,
+      onClick: dissMissOnClick,
+      toggleInputProps,
+      toggleInputComponent,
+      ...passProps
+    } = this.props;
+    const { toggleState } = this.state;
+    const toggleButtonClassName = `${baseClassName}_button`;
+
+    if (toggleInputProps || toggleInputComponent) {
+      logger.error(
+        `ToggleButton does not utilize 'toggleInputProps' and 'toggleInputComponent' props.`,
+      );
+      return false;
+    }
+
+    return (
+      <HtmlButton
+        className={toggleButtonClassName}
```

Comment:
Seem to not be getting this. Should it be that taking `className` from this.props and then using `classnames(toggleButtonClassName, className)`.

I would also like that both toggles get `fi-toggle`, then also `fi-toggle--button` or `fi-toggle--input`.

Reply:
Not sure I understood the latter part correctly, because if I change the ToggleInput to use `fi-toggle--input` instead of previous `fi-toggle_input` it will then show checkbox before the component like in the picture.

<img width="418" alt="Screenshot 2019-10-30 at 14 35 03" src="https://user-images.githubusercontent.com/53757053/67860765-63e8ac80-fb27-11e9-8dd1-6e286d14ea8f.png">

Reply:
`fi-toggle_input` should not be changed. Just that the `span` and `button` would both contain `fi-toggle` but also `fi-toggle--button` for the button and `fi-toggle--input` for the input.

Reply:
This should be now as you described in the recent commit.